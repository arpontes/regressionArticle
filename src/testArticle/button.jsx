import React from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
    root: {
        border: "1px solid blue"
    }
});
export function ButtonWrp({ desc, showBorder }) {
    const classes = useStyles();
    return (
        <div className={showBorder ? classes.root : ""}>
            <Button desc={desc} />
        </div>
    );
}
function Button({ desc }) {
    return <input type="button" value={desc} />;
}
