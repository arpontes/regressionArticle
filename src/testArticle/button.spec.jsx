import React from "react";
import { ButtonWrp } from "./button";
import renderer from 'react-test-renderer';

test('Renderizando botÃ£o', () => {
    const btn = renderer.create(<ButtonWrp desc="Meu Teste" showBorder />).toJSON();
    expect(btn).toMatchSnapshot();
});


