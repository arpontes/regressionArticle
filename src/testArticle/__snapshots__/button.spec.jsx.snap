// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderizando bot√£o 1`] = `
<div
  className="root-0-2-1"
>
  <input
    type="button"
    value="Meu Teste"
  />
</div>
`;
